### PR TITLE
Gs

### PR DESCRIPTION
**Pull request by gsub2000**
I just did the AccountPayment object similar to the Account and Offer APIs, because I wasn't sure if there were any functions unique to this one that had to be done in the backend. My mom told me about how the frontend storing of the "account ID" worked, so I understand that now. 

If I need to add functions to this, just let me know, and I'll start working on it tomorrow. I also want to start researching and just thinking about the Crypto and PaymentMethod objects, but I have a couple questions for those two first, so I'll either email them to you or talk to you about it. 

thanks, Gayatri

### CODE REVIEW

File: src/main/java/com/vaidy/billing/controller/AccountPaymentController.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	@Autowired
+	private AccountPaymentService apService;
+	
+	public AccountPaymentController( AccountPaymentService apService ) {
+		this.apService = apService;
+	}
+	
+	@RequestMapping(
+	
+	value = "/accountpaymentinfo",
```
Comment (jvaidyaa) [inline src/main/java/com/vaidy/billing/controller/AccountPaymentController.java]: let's change this url to createPaymentMethof

File: src/main/java/com/vaidy/billing/controller/AccountPaymentController.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	method = RequestMethod.POST, 
+	consumes = MediaType.APPLICATION_JSON_VALUE,
+	produces = MediaType.APPLICATION_JSON_VALUE
+	) 
+	@ResponseBody()
+	public AccountPayment registerOffer(
+		//@ApiParam("Register Offer Request - This establishes an entity with the system ")
+		@Validated @RequestBody RegisterAccountPaymentReq registerRequest)
+	{
+		AccountPayment accpayment = apService.createAccountPaymentInfo(registerRequest);
```
Comment (jvaidyaa) [inline src/main/java/com/vaidy/billing/controller/AccountPaymentController.java]: likewise change the name of this method too, createPaymentMethod

File: src/main/java/com/vaidy/billing/dto/RegisterAccountPaymentReq.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
+package com.vaidy.billing.dto;
+
```
Comment (jvaidyaa) [inline src/main/java/com/vaidy/billing/dto/RegisterAccountPaymentReq.java]: change the name of this class to CreatePaymentMethodReq

File: src/main/java/com/vaidy/billing/service/AccountPaymentService.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import com.vaidy.billing.entity.AccountPayment;
+import com.vaidy.billing.repository.AccountPaymentRepository;
+import com.vaidy.billing.service.AccountService;
+
+@Service
+
+public class AccountPaymentService {
+
+	@Autowired
+	private AccountPaymentRepository accountPaymentRepository;
```
Comment (jvaidyaa) [inline src/main/java/com/vaidy/billing/service/AccountPaymentService.java:24]: Where is the save to the repo, the paymentMethod does not seemed to be saved

File: src/main/java/com/vaidy/billing/service/AccountPaymentService.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+		accountpayment.setLastName(request.getLastName());
+		accountpayment.setStreet(request.getStreet());
+		accountpayment.setAptSuite(request.getAptSuite());
+		accountpayment.setStreet3(request.getStreet3());
+		accountpayment.setCity(request.getCity());
+		accountpayment.setState(request.getState());
+		accountpayment.setZip(request.getZip());
+		accountpayment.setCountry(request.getCountry());
+		System.out.println(accountpayment.toString());
+
```
Comment (jvaidyaa) [inline src/main/java/com/vaidy/billing/service/AccountPaymentService.java]: We should call the save here

File: src/main/java/com/vaidy/billing/service/AccountPaymentService.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+		return retAccountPayment;		
+		
+	}
+
+
+	public AccountPayment getAccountPaymentInfo(String accountid){
+		return (accountPaymentRepository.findAccountPaymentByAccountid(accountid));
```
Comment (jvaidyaa) [inline src/main/java/com/vaidy/billing/service/AccountPaymentService.java:58]: This method will not be able to get anything as nothing is saved to the db

**Comment by gsub2000** [comment]
I finished fixing all those points

do i have to do another pull request, or..?

**Pull request closed without merging**